src/Effect/DefaultEffector.php

```diff
@@ -40,10 +40,14 @@ public function mergeEffects(string $expr, array $effects, array $results): arra
                 $result = false;
             }
         } elseif ('some(where (p_eft == allow)) && !some(where (p_eft == deny))' == $expr) {
-            $result = true;
+            $result = false;
+
             $explainIndex = array_search(self::DENY, $effects, true);
-            if ($explainIndex !== false) {
-                $result = false;
+            if ($explainIndex === false) {
+                $temp = array_search(self::ALLOW, $effects, true);
+                if ($temp !== false) {
+                    $result = true;
+                }
             }
```

Comment:
@basakest This can be a `for` loop, just one loop, avoid using `array_search` twice .